wagtail/admin/views/generic/models.py

```diff
@@ -596,6 +640,19 @@ def save_instance(self):
         and returns the new object. Override this to implement custom save logic.
         """
         instance = self.form.save()
+        if self.sort_order_field:
+            sort_order_field = getattr(instance, self.sort_order_field)
+            if sort_order_field is None:
+                max_order = (
+                    self.model.objects.aggregate(
+                        max_order=models.Max(self.sort_order_field)
+                    )["max_order"]
+                    or 0
+                )
+                instance_order = max_order + 1
+                instance.__setattr__(self.sort_order_field, instance_order)
+                instance.save(update_fields=[self.sort_order_field])
```

Comment:
Maybe this should be in a separate method so that we don't duplicate the logic in the create/edit optional features mixin.